src/lib.rs

```diff
@@ -39,25 +47,38 @@ pub struct SphinxPacket {
 
 impl SphinxPacket {
     pub fn new(
-        message: Vec<u8>,
+        mut message: Vec<u8>,
         route: &[Node],
         destination: &Destination,
         delays: &[Delay],
-    ) -> Result<SphinxPacket, SphinxUnwrapError> {
+        surb_material: Option<SURBMaterial>,
+    ) -> Result<SphinxPacket, SphinxError> {
         let initial_secret = crypto::generate_secret();
         let (header, payload_keys) =
             header::SphinxHeader::new(initial_secret, route, delays, destination);
 
+        if let Some(surb_material) = surb_material {
+            let surb_initial_secret = crypto::generate_secret();
+            let surb = SURB::new(
+                surb_initial_secret,
+                &surb_material.surb_route,
+                &surb_material.surb_delays,
+                &surb_material.surb_destination,
+            )
+            .unwrap();
+            message = [surb.to_bytes(), message].concat();
```

Comment:
So due to previous comment, you'd perhaps need to clone original message and assign everything to a new variable. Also `concat` might not be the most efficient method of combing vectors due to extra memory allocations, look at `extend` or simple `chain` perhaps?

Reply:
Also we should check if extended message has valid length (since it was extended with a SURB)